test/e2e/token_inactivity_timeout_test.go

```diff
@@ -0,0 +1,284 @@
+	})
+
+	updateOAuthConfigInactivityTimeout(t, configClient, &metav1.Duration{Duration: 600 * time.Second})
+	test.WaitForKubeAPIServerStartProgressing(t, configClient)
+	test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	// With only OAuth config timeout.
+	t.Run("with-inactivity-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, nil, testTokenTimeoutScenarios)
+	})
+
+	// With both OAuth config timeout and OAuthClient timeout.
+	t.Run("with-client-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, &oauthClientTimeout, testTokenTimeoutScenarios)
+	})
+
+	updateOAuthConfigInactivityTimeout(t, configClient, nil)
+	test.WaitForKubeAPIServerStartProgressing(t, configClient)
+	test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	// No OAuthClient timeout and no OAuth config timeout.
+	t.Run("unset-inactivity-timeout-client-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, nil, testTokenTimeoutScenarios)
+	})
+
+}
+
+func checkTokenAccess(t *testing.T,
+	userClient *userclient.UserV1Client,
+	oauthClientClient *oauthclient.OauthV1Client,
+	configInactivityTimeout, oauthClientTimeout *int32,
+	testAccess func(*testing.T, *oauthapi.OAuthAccessToken, *oauthapi.OAuthAccessToken, time.Duration)) {
+	// Create the user, identity, oauthclient and oauthaccesstoken objects needed for authentication using Bearer tokens.
+	subTestNameHierarchy := strings.Split(t.Name(), "/")
+	prefix := subTestNameHierarchy[len(subTestNameHierarchy)-1] + "-"
+
+	userName := prefix + "testuser"
+	idpName := "htpasswd"
+	oauthClientName := prefix + "oauthclient"
+	redirectURIs := []string{"https://localhost"}
+	identityName := idpName + ":" + userName
+
+	expectedInactivityTimeout := *configInactivityTimeout
+	if oauthClientTimeout != nil {
+		expectedInactivityTimeout = *oauthClientTimeout
+	}
+
+	uid, cleanup := createUser(t, userClient, userName, identityName)
+	defer cleanup()
+
+	cleanup = createIdentity(t, userClient, userName, identityName, idpName, uid)
+	defer cleanup()
+
+	cleanup = createOAuthClient(t, oauthClientClient, oauthClientName, redirectURIs, oauthClientTimeout)
+	defer cleanup()
+
+	tokenWithTimeout := &oauthapi.OAuthAccessToken{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: prefix + "token-with-timeout",
+		},
+		ClientName:               oauthClientName,
+		ExpiresIn:                defaultAccessTokenMaxAgeSeconds,
+		Scopes:                   []string{"user:full"},
+		RedirectURI:              redirectURIs[0],
+		UserName:                 userName,
+		UserUID:                  string(uid),
+		AuthorizeToken:           "mJOQ7Es5l9V7WYDl0bvl3E_hRjnJ21ZZxXH6YZj3yeS",
+		InactivityTimeoutSeconds: expectedInactivityTimeout,
```

Comment:
this must be much shorter I think. Would expect something <<120s, such that the use of the token is pushing the timeout beyond 120s. Otherwise the test after 120s will always work, even if `configInactivityTimeout`/`oauthClientTimeout` are not wired correctly into kube-apiserver.

Reply:
Make it 60s and we are fine.

Reply:
made it 60s now.